src/Mod/Part/BOPTools/ShapeMerge.py

```diff
@@ -77,13 +77,18 @@ def splitIntoGroupsBySharing(list_of_shapes, element_extractor, split_connection
         shape_elements.difference_update(split_connections)
         #search if shape is connected to any groups
         connected_to = []
+        not_in_connected_to = []
         for iGroup in range(len(groups)):
             connected = False
             for element in shape_elements:
                 if element in groups[iGroup][1]:
                     connected_to.append(iGroup)
                     connected = True
                     break
+            else:
+                if not connected:
```

Comment:
This line seems useless as induced by the rest of the code.

Reply:
Do you mean `else` part will always be triggered after `connected` is `True`?

Reply:
I mean as it is a `for...else` loop, `else` will be executed only if `break` isn't called, so only if `connected` is `False`

Reply:
You seem correct here. Sorry I mixed up the behavior of `for...else`. I don't seem to recall why this exact structure is here.

Reply:
It wasn't. But it was only a redundant check so no harm.